worlds/sc2/mission_order/presets_static.py

```diff
@@ -592,7 +781,68 @@ def _build_static_preset(preset: Dict[str, Any], options: Dict[str, Any]) -> Dic
                         slot.pop("mission_pool", ())
     else:
         raise ValueError(f"Preset option \"missions\" received unknown value \"{missions}\".")
+    
+    # Key rule selection
+    keys = options.pop("keys", "none")
+    if keys == "layouts":
+        # remove keys from mission entry rules
+        for layout in preset.values():
+            if type(layout) == dict and "missions" in layout:
+                for slot in layout["missions"]:
+                    if "entry_rules" in slot:
+                        slot["entry_rules"] = _remove_key_rules(slot["entry_rules"])
+    elif keys == "missions":
+        # remove keys from layout entry rules
+        for layout in preset.values():
+            if type(layout) == dict and "entry_rules" in layout:
+                layout["entry_rules"] = _remove_key_rules(layout["entry_rules"])
+    elif keys == "none":
+        # remove keys from both layout and mission entry rules
+        for layout in preset.values():
+            if type(layout) == dict:
+                if "entry_rules" in layout:
+                    layout["entry_rules"] = _remove_key_rules(layout["entry_rules"])
+                if "missions" in layout:
+                    for slot in layout["missions"]:
+                        if "entry_rules" in slot:
+                            slot["entry_rules"] = _remove_key_rules(slot["entry_rules"])
+    else:
+        raise ValueError(f"Preset option \"keys\" received unknown value \"{keys}\".")
+    
     return preset
 
+def _remove_key_rules(entry_rules: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    indices: List[int] = []
+    removed = 0
+    for (i, rule) in enumerate(entry_rules):
+        if "items" in rule and "Key" in rule["items"]:
+            indices.append(i)
+    for idx in indices:
+        entry_rules.pop(idx - removed)
+        removed += 1
+    return entry_rules
```

Comment:
As you don't need to change the original list in place (you're using the return value instead), you can do this instead:
```python
return [rule for rule in entry_rules if not ("items" in rule and "Key" in rule["items"])]
```